compass/landice/tests/greenland/mesh.py

```diff
@@ -0,0 +1,209 @@
+        args = ['create_landice_grid_from_generic_MPAS_grid.py',
+                '-i', 'grid_converted.nc',
+                '-o', 'gis_1km_preCull.nc',
+                '-l', levels, '-v', 'glimmer']
+        check_call(args, logger=logger)
+
+        # This step uses a subset of the whole Greenland dataset trimmed to
+        # the region around GIS Glacier, to speed up interpolation.
+        # This could also be replaced with the full Greenland Ice Sheet
+        # dataset.
+        logger.info('calling interpolate_to_mpasli_grid.py')
+        args = ['interpolate_to_mpasli_grid.py', '-s',
+                'greenland_1km_2020_04_20.epsg3413.icesheetonly.nc', '-d',
+                'gis_1km_preCull.nc', '-m', 'b', '-t']
+
+        check_call(args, logger=logger)
+
+        # This step is only necessary if you wish to cull a certain
+        # distance from the ice margin, within the bounds defined by
+        # the GeoJSON file.
+        cullDistance = section.get('cull_distance')
+        if float(cullDistance) > 0.:
+            logger.info('calling define_cullMask.py')
+            args = ['define_cullMask.py', '-f',
+                    'gis_1km_preCull.nc', '-m'
+                    'distance', '-d', cullDistance]
+
+            check_call(args, logger=logger)
+        else:
+            logger.info('cullDistance <= 0 in config file. '
+                        'Will not cull by distance to margin. \n')
+
+        # This step is only necessary because the GeoJSON region
+        # is defined by lat-lon.
+        logger.info('calling set_lat_lon_fields_in_planar_grid.py')
+        args = ['set_lat_lon_fields_in_planar_grid.py', '-f',
+                'gis_1km_preCull.nc', '-p', 'gis-gimp']
+
+        check_call(args, logger=logger)
+
+        dsMesh = xarray.open_dataset('gis_1km_preCull.nc')
+        dsMesh = cull(dsMesh, logger=logger)
+        write_netcdf(dsMesh, 'greenland_culled.nc')
+
+        logger.info('Marking horns for culling')
+        args = ['mark_horns_for_culling.py', '-f', 'greenland_culled.nc']
+        check_call(args, logger=logger)
+
+        logger.info('culling and converting')
+        dsMesh = xarray.open_dataset('greenland_culled.nc')
+        dsMesh = cull(dsMesh, logger=logger)
+        dsMesh = convert(dsMesh, logger=logger)
+        write_netcdf(dsMesh, 'greenland_dehorned.nc')
+
+        logger.info('calling create_landice_grid_from_generic_MPAS_grid.py')
+        args = ['create_landice_grid_from_generic_MPAS_grid.py', '-i',
+                'greenland_dehorned.nc', '-o',
+                'GIS.nc', '-l', levels, '-v', 'glimmer',
+                '--beta', '--thermal', '--obs', '--diri']
+
+        check_call(args, logger=logger)
+
+        logger.info('calling interpolate_to_mpasli_grid.py')
+        args = ['interpolate_to_mpasli_grid.py', '-s',
+                'greenland_1km_2020_04_20.epsg3413.icesheetonly.nc',
+                '-d', 'GIS.nc', '-m', 'b', '-t']
```

Comment:
Should we interpolate all available fields (e.g. SMB, velo obs) in this final step?

Reply:
Updated in https://github.com/MPAS-Dev/compass/pull/351/commits/03f0a464438e68a99d5e485c36a22951946dacc3. Note that basalHeatFlux ends up with the wrong sign because of the inconsistent conventions, so I will update the gridded dataset on the database.